### PR TITLE
Fix fullscreenState toggling behaviour

### PR DESCRIPTION
**Pull request by txkyel**
#### Describe your PR, what does it fix/add?

When the user toggles the either the `internal` or `client` fullscreen states (using `-1` to not alter the other state) the relevant state is toggled. 

e.g. when the initial state is `1 2` and `1 -1` is dispatched, the state is now set to `0 2` instead of syncing the states to `2 2`.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

None

#### Is it ready for merging, or does it need work?

Ready for merge

fyi @MightyPlaza

Thread on previous commit relevant to changes: https://github.com/hyprwm/Hyprland/commit/2b520571e897be2a0e88c8692da607b062000038#r145242308

### CODE REVIEW

**Comment by txkyel** [comment]
I'd like to add documentation for this behaviour but I'm not too sure whether it'd be more appropriate to place on the dispatchers page or the rules page. 

Any suggestions?

**Comment by MightyPlaza** [comment]
I don't think setting `syncFullscreen` default to false is a good idea, in most cases you'll want it set to true (for example when using a fullscreen button on a client itself)

File: src/Compositor.cpp [hunk 1]
```diff
@@ -2305,10 +2305,6 @@ void CCompositor::setWindowFullscreenState(const PHLWINDOW PWINDOW, sFullscreenS
 
     const bool            CHANGEINTERNAL = !(PWINDOW->m_bPinned || CURRENT_EFFECTIVE_MODE == EFFECTIVE_MODE || (PWORKSPACE->m_bHasFullscreenWindow && !PWINDOW->isFullscreen()));
 
-    // TODO: update the state on syncFullscreen changes
-    if (!CHANGEINTERNAL && PWINDOW->m_sWindowData.syncFullscreen.valueOrDefault())
```
Comment (MightyPlaza) [inline src/Compositor.cpp:2309]: this is needed to check if to parse the current state when `syncFullscreen` is set to true

**Comment by txkyel** [comment]
It looks like I've completely misunderstood the functionality and importants of `syncFullscreen`. I expected `syncFullscreen` to only affect the behaviour of the `fullscreenState` dispatcher. 

I still don't quite agree with the toggle behaviour of the `fullscreenState`. Perhaps it would instead make sense to add an additional parameter determine the toggle behaviour to toggle the individual state or to sync the states. What do you think?

**Comment by MightyPlaza** [comment]
> It looks like I've completely misunderstood the functionality and importants of `syncFullscreen`. I expected `syncFullscreen` to only affect the behaviour of the `fullscreenState` dispatcher. I'll revert these changes.
> 
> I still don't quite agree with the toggle behaviour of the `fullscreenState`. Perhaps it would instead make sense to add an additional parameter determine the toggle behaviour to toggle the individual state or to sync the states. What do you think?

I think the best solution for now would be to toggle it to `FSMODE_NONE` individually and only re-sync if both modes get toggled to `FSMODE_NONE`.

**Comment by txkyel** [comment]
> re-sync if both modes get toggled to FSMODE_NONE

Do you mean we should set `syncFullscreen` to false in the cases where only one state is toggled? What would be the effect of leaving `syncFullscreen` as true when the internal and client states are not in sync?

**Comment by MightyPlaza** [comment]
> > re-sync if both modes get toggled to FSMODE_NONE
> 
> Do you mean we should set `syncFullscreen` to false in the cases where only one state is toggled? What would be the effect of leaving `syncFullscreen` as true when the internal and client states are not in sync?

I mean when toggling back to `FSMODE_NONE` set `syncFullscreen` to true, to fully toggle the behaviour of the dispatcher

when `syncFullscreen` is set to true the internal and client modes should be kept the same

File: src/managers/KeybindManager.cpp [hunk 2]
```diff
@@ -1181,14 +1181,14 @@ void CKeybindManager::fullscreenStateActive(std::string args) {
     }
 
     if (internalMode != -1 && clientMode == -1 && PWINDOW->m_sFullscreenState.internal == STATE.internal) {
-        g_pCompositor->setWindowFullscreenState(PWINDOW, sFullscreenState{.internal = PWINDOW->m_sFullscreenState.client, .client = PWINDOW->m_sFullscreenState.client});
-        PWINDOW->m_sWindowData.syncFullscreen = CWindowOverridableVar(true, PRIORITY_SET_PROP);
+        g_pCompositor->setWindowFullscreenState(PWINDOW, sFullscreenState{.internal = FSMODE_NONE, .client = PWINDOW->m_sFullscreenState.client});
+        PWINDOW->m_sWindowData.syncFullscreen = CWindowOverridableVar(false, PRIORITY_SET_PROP);
```
Comment (MightyPlaza) [inline src/managers/KeybindManager.cpp:1185]: `syncFullscreen` should be set to false above

**Review by vaxerski — APPROVED** [review]

**Pull request merged**